sdk/personalizer/Azure.AI.Personalizer/src/Models/DecisionContext.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Azure.AI.Personalizer
+{
+    /// <summary> The Decision Context. </summary>
+    public class DecisionContext
+    {
+        /// <summary> The Decision Context used to serialize an object. </summary>
+        public DecisionContext()
+        {
+        }
+
+        /// <summary> Initializes a new instance of DecisionContext. </summary>
+        /// <param name="rankRequest"> Personalizer Rank Options </param>
+        public DecisionContext(PersonalizerRankOptions rankRequest)
+        {
+            List<string> jsonFeatures = rankRequest.ContextFeatures.Select(f => JsonSerializer.Serialize(f)).ToList();
+            this.SharedFromUrl = jsonFeatures;
+            HashSet<string> excludedSet = new HashSet<string>(rankRequest.ExcludedActions);
+
+            IEnumerable<PersonalizerRankableAction> rankableActions = rankRequest.Actions.Where(action => !excludedSet.Contains(action.Id));
+            this.Documents = rankableActions
+                .Select(action =>
+                {
+                    List<string> jsonFeatures = action.Features.Select(f => JsonSerializer.Serialize(f)).ToList();
+
+                    var doc = new DecisionContextDocument
+                    {
+                        ID = action.Id,
+                        JSON = jsonFeatures,
+                    };
+
+                    return doc;
+                }).ToArray();
+        }
+
+        /// <summary> Properties from url </summary>
+        [JsonPropertyName("FromUrl")]
+        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+        [JsonConverter(typeof(JsonRawStringListConverter))]
+        public List<string> SharedFromUrl { get; }
```

Comment:
"FromUrl" is the default context feature namespace we add. This namespace contains all context features which are passed in by the user. I think this property should be called ContextFeatures

Reply:
agreed